### ISSUE TITLE
Revision control adding 'v' to remote versions

### ISSUE THREAD

**Issue by info-rchitect** [Origen-SDK/origen#229]
Hi,

The Origen revision control module (in this case Git) seems to be pre-pending a 'v' to the remote version.  I tagged my remote repo as '0.1.0' and it failed.  Here is my remote config:

~~~ruby
  config.remotes = [
    {
      dir: 'product',
      vault: 'ssh://git@stash.mycompany.com:7999/myproduct/origen.git',
      version: '0.1.0'
    }
  ]
~~~

Here is the error:

~~~shell
peologin02:ppekit $ rm -rf ../.ppekit_remotes_DO_NOT_HAND_MODIFY/
peologin02:ppekit $ origen i
[INFO]       0.017[0.017]    || *******************************************************************************
[INFO]       0.017[0.000]    || The following remotes need to be updated, this will now happen automatically:
[INFO]       0.018[0.001]    ||   product - 0.1.0 (included by ppekit)
[INFO]       0.018[0.000]    ||
[INFO]       0.056[0.000]    || git init
[INFO]       0.056[0.000]    ||
[INFO]       1.340[1.284]    || Initialized empty Git repository in /users/user/origen/.ppekit_remotes_DO_NOT_HAND_MODIFY/product/.git/
[INFO]       1.361[0.021]    || git remote add origin ssh://git@stash.mycompany.com:7999/myproduct/origen.git
[INFO]       1.361[0.000]    ||
[INFO]       1.389[0.028]    || git fetch
[INFO]       1.389[0.000]    ||
[INFO]       5.520[4.131]    || From ssh://stash.mycompany.com:7999/myproduct/origen
[INFO]       5.629[0.095]    || * [new branch]      master     -> origin/master
[INFO]       5.637[0.007]    || * [new tag]         0.1.0      -> 0.1.0
[INFO]       5.892[0.255]    || git reset --hard v0.1.0
[INFO]       5.892[0.000]    ||
[INFO]       5.912[0.020]    || fatal: ambiguous argument 'v0.1.0': unknown revision or path not in the working tree.
[INFO]       5.912[0.000]    || Use '--' to separate paths from revisions, like this:
[INFO]       5.913[0.000]    || 'git <command> [<revision>...] -- [<file>...]'

COMPLETE CALL STACK
-------------------
When updating remotes for ppekit: This command failed: 'git reset --hard v0.1.0'
~~~

If I change my stash repo tag to 'v0.1.0' it works.  Should Origen be forcing tag syntax?

thx

**Comment by pderouen** [comment]
I don't know much about the revision control object.  But, from what you describe it seems like you told origen to populate a specific version and it added a leading 'v' to the version you provided.  Is that what happened?

**Comment by info-rchitect** [comment]
@pderouen yes that is what appears to be happening.  If this is an NXP specific thing, could someone highlight what needs to be changed?  I can handle the PR.

**Comment by priyavadan** [comment]
@info-rchitect I believe it might be related to this https://github.com/Origen-SDK/origen/blob/master/lib/origen/version_string.rb#L354 ?

**Comment by pderouen** [comment]
"prefix_tag" in remote_manager.rb has a comment above it saying that it will add the leading 'v' if the version tag looks like a semantic version number.  Seems like this was intentionally done.

**Comment by pderouen** [comment]
https://github.com/Origen-SDK/origen/blob/8e59a0dd340f8c61bf0268673e52b5463159572b/lib/origen/remote_manager.rb#L340

**Comment by info-rchitect** [comment]
thanks @pderouen  and @priyavadan.  I checked the code and agree with your conclusion this is by design.    While I think this is a good concept, it is not practical for an industry wide solution.  Our tags have multiple other tools/systems designed to read a specific version config so we can't change this practically.  Any way we can make this an application specific option in the application.rb file, similar to:

~~~ruby
config.rc.tag_prepend = 'v'
~~~

Or allow us to opt out with something similar:

~~~ruby
config.rc.tag_prepend = nil
~~~

That way all of the existing apps wouldn't have to change, our apps would change.

thx

**Comment by pderouen** [comment]
Sounds good to me.  I'm not sure why the 'v' is forced.  I'm sure there was a good reason.  But, there should be a way to opt out.

**Comment by ginty** [comment]
The v is forced because designsync will not accept a tag that starts with a
number.
I'm cool with whatever solution you come up with, though it has to be
backwards compatible.
I would recommend that you add a deprecation warning if you use the
existing API, saying that the v will no longer be prepended in origen v1.
In the short term though you will need to add some way of turning off the
prepend at application level.

On 21 Feb 2018 19:58, "pderouen" <notifications@github.com> wrote:

Sounds good to me. I'm not sure why the 'v' is forced. I'm sure there was a
good reason. But, there should be a way to opt out.

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub
<https://github.com/Origen-SDK/origen/issues/229#issuecomment-367432765>,
or mute the thread

**Comment by ginty** [comment]
What about introducing `:tag` as a remote option, that would be used verbatim, whereas the `:version` attribute would be subject to the prepend when converting to a tag.

**Comment by info-rchitect** [comment]
@ginty i like your solution.  we will handle this in due time.  for our app we ended up just using raw git commands in boot.rb to get what we needed currently.

**Comment by info-rchitect** [comment]
completed [here](https://github.com/Origen-SDK/origen/pull/243)
